client/core/trade.go

```diff
@@ -1185,11 +1215,35 @@ func (c *Core) swapMatches(t *trackedTrade, matches []*matchTracker) error {
 	}
 	receipts, change, fees, err := t.wallets.fromWallet.Swap(swaps)
 	if err != nil {
-		// Set the error on the matches.
 		for _, match := range matches {
-			match.failErr = err
+			// Mark the matches as suspect to prevent them being grouped again.
+			match.suspectSwap = true
+			// If we can still swap before the broadcast timeout, allow retries
+			// soon.
+			lastActionTime := match.matchTime()
+			if match.Match.Side == order.Taker {
+				// It is possible that AuditStamp could be zero if we're
+				// recovering during startup. The implications of this is that
+				// if we are 1) already in start up recovery, and 2) we fail
+				// again, we will not try again even if we have time before the
+				// broadcast timeout.
+				lastActionTime = encode.UnixTimeMilli(int64(match.MetaData.Proof.Auth.AuditStamp))
+			}
+			if time.Since(lastActionTime) < t.broadcastTimeout() {
+				m := match // Scoped for the AfterFunc.
+				match.tickMeterer = time.AfterFunc(t.dc.tickInterval*3/4, func() {
+					t.mtx.Lock()
+					defer t.mtx.Unlock()
+					m.tickMeterer = nil
+				})
+			} else {
+				// If we can't get a swap out before the broadcast timeout, just
+				// quit.
+				match.failErr = err
```

Comment:
I think the `failErr` field needs a description of it's purpose (a sentinel to stop repeated attempts to swap when there is there is no longer time to act?).  No longer set for redeems ever it seems, right?